packages/cli/src/requests.ts

```diff
@@ -322,6 +322,13 @@ export declare namespace UserRequest {
 		{ transferId?: string; includeRole: boolean }
 	>;
 
+	export type ChangeRole = AuthenticatedRequest<
+		{ id: string },
+		{},
+		{ newRole?: { scope?: RoleScopes; name?: RoleNames } },
```

Comment:
Maybe I'm just missing something but if it's `ChangeRole` request what would it do without the `newRole` object?
Why is it optional?

Reply:
Or is it just to keep the contract real-life like? Like I can send any bullshit request

Reply:
> Or is it just to keep the contract real-life like?

Exactly! If we don't do this, when working in the controller we'll assume (based on the typing) that this property is guaranteed to be there, which is not the case and requires validation.